hrqb/tasks/employee_salary_history.py

```diff
@@ -0,0 +1,92 @@
+"""hrqb.tasks.employee_salary_history"""
+
+import luigi  # type: ignore[import-untyped]
+import pandas as pd
+
+from hrqb.base.task import PandasPickleTask, QuickbaseUpsertTask, SQLQueryExtractTask
+from hrqb.utils import normalize_dataframe_dates
+
+
+class ExtractDWEmployeeSalaryHistory(SQLQueryExtractTask):
+    """Query Data Warehouse for employee salary history data."""
+
+    stage = luigi.Parameter("Extract")
+
+    @property
+    def sql_file(self) -> str:
+        return "hrqb/tasks/sql/employee_salary_history.sql"
+
+
+class TransformEmployeeSalaryHistory(PandasPickleTask):
+    stage = luigi.Parameter("Transform")
+
+    def requires(self) -> list[luigi.Task]:  # pragma: nocover
+        from hrqb.tasks.shared import ExtractQBEmployeeAppointments
+
+        return [
+            ExtractDWEmployeeSalaryHistory(pipeline=self.pipeline),
+            ExtractQBEmployeeAppointments(pipeline=self.pipeline),
+        ]
+
+    def get_dataframe(self) -> pd.DataFrame:
+        dw_salary_df = self.named_inputs["ExtractDWEmployeeSalaryHistory"].read()
+        qb_emp_appts_df = self.named_inputs["ExtractQBEmployeeAppointments"].read()
```

Comment:
Anticipating some abbrevation comments here...

After some work in these tasks, my feeling is that the trade-off is worth it vs really cumbersome variables like `quickbase_employee_appointments_df`.